netbox/extras/views.py

```diff
@@ -1042,116 +1042,177 @@ def get_script_module(module, request):
     return get_object_or_404(ScriptModule.objects.restrict(request.user), file_path__regex=f"^{module}\\.")
 
 
-class ScriptView(ContentTypePermissionRequiredMixin, View):
+class BaseScriptView(ContentTypePermissionRequiredMixin, generic.ObjectView):
```

Comment:
These views should be extended to support full object-based permissions now that scripts can be referenced as database objects